sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/plans/joinTypes.scala

```diff
@@ -100,7 +100,7 @@ case class NaturalJoin(tpe: JoinType) extends JoinType {
   override def sql: String = "NATURAL " + tpe.sql
 }
 
-case class UsingJoin(tpe: JoinType, usingColumns: Seq[UnresolvedAttribute]) extends JoinType {
```

Comment:
Yeah, we do not support the nested field. It also fails with your newly changed error. 
```
    sql("CREATE TABLE complexTypeTable (s struct<i: string>)")
    val df = table("complexTypeTable")
    df.as("b").join(df.as("a"), "s.i").show()
```

Could you add the test case for it? Thanks!